packages/shared/tests/normalizeRoutePath.spec.ts

```diff
@@ -2,109 +2,232 @@ import { describe, expect, it } from 'vitest'
+  [['./foo/bar.html', '/a/index.html'], '/a/foo/bar.html'],
+  [['./foo/bar.html', '/a/b.html'], '/a/foo/bar.html'],
+  [['../foo', '/a/'], '/foo.html'],
+  [['../foo', '/a/index.html'], '/foo.html'],
+  [['../foo', '/a/b.html'], '/foo.html'],
+  [['../foo.md', '/a/'], '/foo.html'],
+  [['../foo.md', '/a/index.html'], '/foo.html'],
+  [['../foo.md', '/a/b.html'], '/foo.html'],
+  [['../foo.html', '/a/'], '/foo.html'],
+  [['../foo.html', '/a/index.html'], '/foo.html'],
+  [['../foo.html', '/a/b.html'], '/foo.html'],
+  [['../foo/bar', '/a/'], '/foo/bar.html'],
+  [['../foo/bar', '/a/index.html'], '/foo/bar.html'],
+  [['../foo/bar', '/a/b.html'], '/foo/bar.html'],
+  [['../foo/bar.md', '/a/'], '/foo/bar.html'],
+  [['../foo/bar.md', '/a/index.html'], '/foo/bar.html'],
+  [['../foo/bar.md', '/a/b.html'], '/foo/bar.html'],
+  [['../foo/bar.html', '/a/'], '/foo/bar.html'],
+  [['../foo/bar.html', '/a/index.html'], '/foo/bar.html'],
+  [['../foo/bar.html', '/a/b.html'], '/foo/bar.html'],
+
+  // absolute non index with current
+  [['/foo', '/'], '/foo.html'],
+  [['/foo', '/a.html'], '/foo.html'],
+  [['/foo', '/index.html'], '/foo.html'],
+  [['/foo', '/a/'], '/foo.html'],
+  [['/foo', '/a/index.html'], '/foo.html'],
+  [['/foo', '/a/b.html'], '/foo.html'],
+  [['/foo.md', '/'], '/foo.html'],
+  [['/foo.md', '/a.html'], '/foo.html'],
+  [['/foo.md', '/index.html'], '/foo.html'],
+  [['/foo.md', '/a/'], '/foo.html'],
+  [['/foo.md', '/a/index.html'], '/foo.html'],
+  [['/foo.md', '/a/b.html'], '/foo.html'],
+  [['/foo.html', '/'], '/foo.html'],
+  [['/foo.html', '/a.html'], '/foo.html'],
+  [['/foo.html', '/index.html'], '/foo.html'],
+  [['/foo.html', '/a/'], '/foo.html'],
+  [['/foo.html', '/a/index.html'], '/foo.html'],
+  [['/foo.html', '/a/b.html'], '/foo.html'],
+  [['/foo/bar', '/'], '/foo/bar.html'],
+  [['/foo/bar', '/a.html'], '/foo/bar.html'],
+  [['/foo/bar', '/index.html'], '/foo/bar.html'],
+  [['/foo/bar', '/a/'], '/foo/bar.html'],
+  [['/foo/bar', '/a/index.html'], '/foo/bar.html'],
+  [['/foo/bar', '/a/b.html'], '/foo/bar.html'],
+  [['/foo/bar.md', '/'], '/foo/bar.html'],
+  [['/foo/bar.md', '/a.html'], '/foo/bar.html'],
+  [['/foo/bar.md', '/index.html'], '/foo/bar.html'],
+  [['/foo/bar.md', '/a/'], '/foo/bar.html'],
+  [['/foo/bar.md', '/a/index.html'], '/foo/bar.html'],
+  [['/foo/bar.md', '/a/b.html'], '/foo/bar.html'],
+  [['/foo/bar.html', '/'], '/foo/bar.html'],
+  [['/foo/bar.html', '/a.html'], '/foo/bar.html'],
+  [['/foo/bar.html', '/index.html'], '/foo/bar.html'],
+  [['/foo/bar.html', '/a/'], '/foo/bar.html'],
+  [['/foo/bar.html', '/a/index.html'], '/foo/bar.html'],
+  [['/foo/bar.html', '/a/b.html'], '/foo/bar.html'],
+
+  // only hash and query
+  [[''], ''],
 
   // unexpected corner cases
-  ['.md', '.html'],
-  ['foo/.md', 'foo/.html'],
-  ['/.md', '/.html'],
-  ['/foo/.md', '/foo/.html'],
+  [['.md'], '.html'],
+  [['foo/.md'], 'foo/.html'],
+  [['/.md'], '/.html'],
+  [['/foo/.md'], '/foo/.html'],
```

Comment:
Still need cases with current for the last 5 lines

Reply:
I actually don't think this is needed,  These are corner cases which we are not expecting users to actually set them in production

Reply:
They are unexpected, but still should be covered by our unit tests

Reply:
For example, if a method is going to throw an error with unexpected input, we usually test it with `expect().toThrow()`. For this case, we don't throw errors, but we still need to be aware of those unexpected input.